### ISSUE TITLE
Run a single index update loop thread

### ISSUE THREAD

**Issue by gterzian** [meilisearch/MeiliSearch#1197]
**Is your feature request related to a problem? Please describe.**
When a new database is created, an `update_awaiter` loop is started in a thread per index:

https://github.com/meilisearch/MeiliSearch/blob/08a8dc0d0d0f4872878ec85a7e6735ed694f44ff/meilisearch-core/src/database.rs#L289

Is there a reason for having one thread/updater per index? I assume the underlying database does not run transactions in parallel, and that thus more threads does not give more parallelism, while still generating some overhead per index. 

**Describe the solution you'd like**
Have you considered spawning a single "updater" thread? 

It could own a map of index id to whatever info is required to update an index(probably what is currently being passed as argument to `update_awaiter`, and perhaps only one ref to things like the "env" would be necessary). 

The update events would have to contain the id of the index to which they relate. 

The single update loop would then receive messages, look up the "updater" in the map using the id in the message, and run the update logic for the index. 

The benefits could be:

1. Using only a single thread(less memory, maybe better cache locality, less context switching and so on).
2. A more cohesive "updater" component.

**Comment by MarinPostma** [comment]
Hello @gterzian, thanks for the suggestions, we're working on a new version of the engine that handle this very differently. We won't fix this in the current version.

cheers!
